lib/ts/recipe/dashboard/api/implementation.ts

```diff
@@ -33,6 +33,8 @@ export default function getAPIImplementation(): APIInterface {
             let connectionURI: string = "";
             const superTokensInstance = SuperTokens.getInstanceOrThrowError();
 
+            const authMode: AuthMode = input.options.config.apiKey ? "api-key" : "email-password";
```

Comment:
You could change recipe config normalisation to add a key to it for this. Will be useful if you need to make this check in other places as well